src/allotropy/parsers/nexcelom_matrix/nexcelom_matrix_parser.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import Any
+
+import pandas as pd
+
+from allotropy.allotrope.models.cell_counting_benchling_2023_11_cell_counting import (
+    CellCountingAggregateDocument,
+    CellCountingDetectorDeviceControlAggregateDocument,
+    CellCountingDetectorMeasurementDocumentItem,
+    CellCountingDocumentItem,
+    DataSystemDocument,
+    DeviceControlDocumentItemModel,
+    MeasurementAggregateDocument,
+    Model,
+    ProcessedDataAggregateDocument1,
+    ProcessedDataDocumentItem,
+    SampleDocument,
+)
+from allotropy.allotrope.models.shared.definitions.custom import (
+    TQuantityValueCell,
+    TQuantityValueMicrometer,
+    TQuantityValueMillionCellsPerMilliliter,
+    TQuantityValuePercent,
+)
+from allotropy.constants import ASM_CONVERTER_NAME, ASM_CONVERTER_VERSION
+from allotropy.exceptions import AllotropeConversionError
+from allotropy.named_file_contents import NamedFileContents
+from allotropy.parsers.nexcelom_matrix.constants import EPOCH_STR
+from allotropy.parsers.nexcelom_matrix.nexcelom_matrix_reader import (
+    NexcelomMatrixReader,
+)
+from allotropy.parsers.utils.uuids import random_uuid_str
+from allotropy.parsers.vendor_parser import VendorParser
+
+
+class SampleProperty(Enum):
+    LIVE_COUNT = ("Live Count", TQuantityValueCell)
+    LIVE_MEAN_SIZE = ("Live Mean Size", TQuantityValueMicrometer)
+    DEAD_COUNT = ("Dead Count", TQuantityValueCell)
+    DEAD_CELLS_ML = ("Dead Cells/mL", TQuantityValueMillionCellsPerMilliliter)
+    DEAD_MEAN_SIZE = ("Dead Mean Size", TQuantityValueMicrometer)
+    TOTAL_COUNT = ("Total Count", TQuantityValueCell)
+    TOTAL_CELLS_ML = ("Total Cells/mL", TQuantityValueMillionCellsPerMilliliter)
+    TOTAL_MEAN_SIZE = ("Total Mean Size", TQuantityValueMicrometer)
+
+    def __init__(self, column_name: str, data_type: Any) -> None:
+        self.column_name: str = column_name
+        self.data_type: Any = data_type
+
+
+def get_property_from_sample(
+    sample: pd.Series[Any], sample_property: SampleProperty
+) -> Any:
+    value = sample.get(sample_property.column_name)
+    return sample_property.data_type(value=value) if value else None
+
+
+class NexcelomMatrixParser(VendorParser):
+    def to_allotrope(self, named_file_contents: NamedFileContents) -> Model:
+        contents = named_file_contents.contents
+        filename = named_file_contents.original_file_name
+        filepath = contents.name
+        reader = NexcelomMatrixReader(contents)
+
+        return Model(
+            field_asm_manifest="http://purl.allotrope.org/manifests/cell-counting/BENCHLING/2023/11/cell-counting.manifest",
+            cell_counting_aggregate_document=CellCountingAggregateDocument(
+                data_system_document=DataSystemDocument(
+                    file_name=filename,
+                    UNC_path=filepath,
+                    ASM_converter_name=ASM_CONVERTER_NAME,
```

Comment:
```suggestion
                    ASM_converter_name=self.asm_converter_name,
```
(this is a change since you started this)